wip: Make platform multi-language and themeable

frequent opened the pull request:
I forked the Abstimmungsplatform trying to adapt it for other movements, namely [Volteuropa](https://github.com/VoltEuropa) whom I'm volunteering for. Our Mike Szyndel and your Clemens Holtmann were in touch.

In a first step, I tried to add multi-language support and make the platform theme-able:
1) multi-language 
= add translation flags and use .po files only (!) for translations
= allow to select and set language when logged in and out
2) theme-able
= move DiB specific content (links, terms) into an [init](https://github.com/frequent/abstimmungstool/blob/i18n/init.ini) file, fetch parameters via configparser
= make the de-DE .po file "DiB specific" - it (hopefully) has all the Beweger*innen etc texts

I did not find any tests to try the changes against and did not really manage to see through whether everything works. I'd be interested in doing a test scenario using the [example_data.json](https://github.com/DemokratieInBewegung/abstimmungstool/blob/master/example_data.json) or a derivative from it, so if you can point me to:
- are there any tests your run to verify changes work?
- how does the example data get into the platform?

I'd try and write tests which also work for the current state of the platform.

gnunicorn commented on templates/fragments/argument/item.html:
```diff
@@ -1,81 +1,171 @@
-			Weniger anzeigen
-			</a>
-			{% endif %}
+chronos@localhost /usr/local/demo/abstimmungstool $ cat templates/fragments/argument/item.html
```
I am pretty sure this line doesn't belong in here ;) .

frequent replied:
pretty sure, too :)

gnunicorn commented on templates/fragments/initiative/list.html:
```diff
@@ -1,15 +1,19 @@
-</script>
\ No newline at end of file
+// XXX really necessary?
+(function (window)  {
```
this change is unrelated to this PR.

gnunicorn commented on templates/initproc/blocks/magic_search.html:
```diff
@@ -1,43 +0,0 @@
-{% load static %}
```
Why are you removing all of our magic search? It is only temporarily deactivated until we've figured out a decent UI to give to users...

frequent replied:
I ran into code here and there wondering whether it was actually being used (multiple submit buttons if you want another example). As I had no info, I remove some files, but I'll add them back again now. thanks for explaining.

gnunicorn commented on voty/initadmin/templatetags/site_defaults.py:
```diff
@@ -0,0 +1,35 @@
+from six.moves import configparser
+
+site_parser = configparser.ConfigParser()
+site_parser.read(os.path.join(settings.BASE_DIR, "init.ini"))
```
I don't understand: why have a second configuration file if there is already the `settings.py`? Wouldn't it be much easier to provide another configuration file and import it with a `from local_config import *` in `settings.py` and then use the more generic `get_settings` you provided below? Why did you opt for this approach?

frequent replied:
because voty should stay voty, no matter what initial configuration is fed into. see custom roles/permissions I currently did. The `settings.py` should ... "set" the voty app based on a configuration file loaded. So anyone wanting to use voty can make his own `init.ini`.

gnunicorn commented on voty/initadmin/views.py:
```diff
@@ -74,58 +75,62 @@ def invite_em(file):
+# active users (recently logged in first)
+@login_required
+@user_passes_test(lambda u: u.is_staff)
+def active_users(request):
```
this change seems unrelated to the i18n.

gnunicorn commented on voty/initadmin/urls.py:
```diff
@@ -1,13 +1,10 @@
-    url('^backoffice/download_csv/(?P<id>.*)$', views.download_csv, name='download_mass_invites'),
-]
\ No newline at end of file
+  url(r"^account/edit$", views.profile_edit, name="profile_edit"),
```
these changes seems unrelated to the i18n.

gnunicorn commented on voty/initproc/globals.py:
```diff
@@ -65,17 +65,17 @@ class VOTED:
 ]
 
 SUBJECT_CATEGORIES = [
-    'Globale Politik & internationale Zusammenarbeit',
```
now this one becomes tricky, as those are stored as database-values and therefore _do not_ depend on the users i18n settings. At least not here, but only when displayed. We've to investigate further how to make this happen properly.

frequent replied:
I think voty does cover all cases of where to put text in places rather difficult to catch and translate ;)

gnunicorn commented on voty/initproc/migrations/0025_auto_20180623_1619.py:
```diff
@@ -0,0 +1,30 @@
+# -*- coding: utf-8 -*-
```
yeah, this is the storing of values I was talking about. We can't really do/merge it like this...

frequent replied:
agree :)

gnunicorn commented on voty/initproc/models.py:
```diff
@@ -346,8 +356,10 @@ class Quorum(models.Model):
     def current_quorum(cls):
-        return cls.objects.order_by("-created_at").values("quorum").first()["quorum"]
-
+        quorum_list = cls.objects.order_by("-created_at").values("quorum")
```
unrelated change.

Sidenote: also, as there should generally be a quorum (the task should be run regularly), if anything it shouldn't query the DB twice (which `len(..)` and `first()` do, but rather have a `try: return ...first(); except: return 0`-single-query-fallback.

gnunicorn commented on voty/initproc/urls.py:
```diff
@@ -3,29 +3,36 @@
+  url(r"^$", views.index, name="home"),
+
+  # about (ueber) is not static
+  url(r"^about/$", views.about, name="about"),
```
this also get tricky as we do want i18n' URLs, but system-wide translated rather than user-focussed (so our system would want a german url, even if you are using the engl. edition). I think this is possible (using the default-locale) but need to investigate more.

frequent replied:
I've run into this a few days ago when trying to link from a notification to a user/initiative. Pinax translates the url by default, so I had to add the first ["language agnostic"](https://github.com/VoltEuropa/abstimmungstool/blob/volt/voty/initadmin/urls.py#L41) urls to  `urls.py`. I also need to check more how to do this for all urls.

gnunicorn commented on voty/settings.py:
```diff
@@ -12,25 +12,27 @@
 
+# Retrieve initialization configuration
+init_parser = configparser.ConfigParser()
+init_parser.read(os.path.join(BASE_DIR, 'init.ini'))
```
as above: why not just have/use a second `local_config.py` and do a `from local_config import *` at the end of the file? Why have to different configuration files in different formats (one python, the other as ini)?

gnunicorn commented on voty/settings.py:
```diff
@@ -158,33 +161,29 @@
 USE_TZ = True
 
+LOCALE_PATHS = (
+  os.path.join( BASE_DIR, 'locale'),
```
how about adding another `locale` dir here for customisations? This way, we could provide generic translations (without the term "Beweger") and every instance could easily customize their language..?

(We could later extend the usage of that folder to also include customising of styles and alike)

frequent replied:
This sounds good.

gnunicorn commented:
> are there any tests your run to verify changes work?

Unfortunately not, no :( . However, we have a staging environment we run our latest develop-version on, so we'd notice errors hopefully long before we move it to production (worked fin-ish enough for now).

>  how does the example data get into the platform?

`python manage.py loaddata example_data.json`

However, I am not sure this batch of data is up to date with the latest migrations...

frequent commented:
@gnunicorn thanks for replying. Give me a few days to reply, it's been a while and I have to puzzle stuff back together. I'll report back in a bit.

frequent commented:
@gnunicorn : finally a bit of time to answer a bit more in detail.

Backstory in a nutshell:
- from (Volt) side I volunteered to try extend your voty tool to what our policy team had in mind.
- while I do work in software development, I never did anything in Django
- thus translation was more of an idea to get an overview and grasp
- I needed a while to get an unmodified clone of your master-branch to run without errors 
- I added users/initiatives but still can't get through the life cycle from draft to accepted, can't open for discussion)

Current status:
so after doing the MR (which I guess was more of ping if you want :) I've switched to [another branch](https://github.com/VoltEuropa/abstimmungstool/tree/volt) and decided to first implement roles/permissions/expand initadmin, so that our policy team can eventually manage the tool without having to touch the admin-panel and without having to ping our tech team. This includes:
- configurable [groups and permissions](https://github.com/VoltEuropa/abstimmungstool/blob/volt/init.ini#L61)
- populating [scope](https://github.com/VoltEuropa/abstimmungstool/blob/volt/init.ini#L175) (your "Ebene") for later filtering users/initiatives
- switching [notifications to more of an info/todolist](https://github.com/VoltEuropa/abstimmungstool/blob/volt/init.ini#L98) and moved texts to the init file
- I added a [user list](https://github.com/VoltEuropa/abstimmungstool/blob/volt/voty/initadmin/views.py#L213) for moderating users
- I couldn't get batch registration/batch signup to work, so I [redid the whole section](https://github.com/VoltEuropa/abstimmungstool/blob/volt/voty/initadmin/views.py#L333) them
- I added a [moderator menu](https://github.com/VoltEuropa/abstimmungstool/blob/volt/voty/initadmin/views.py#L390) to reset user email, scope, groups, deactivate/delete/anonymise user
- I [bundled settings](https://github.com/VoltEuropa/abstimmungstool/blob/volt/voty/initadmin/views.py#L530), fixed notifications/avatar menu, added preferred language and localisation

I'll try to merge your latest changes on top, it seems they are mostly in initproc, so in theory there shouldn't be any conflicts :) I wanted to have a rough demo for our Policy Team ready in a week before going on holidy and since I still can't get an initiative through all phases, I'll likely continue "sewing with a hot needle" and not worry about how to align the two votys properly. End of August I'll have more time and then we should discuss how to keep things moving in parallel - I spoke on this with your Clemens and it's also what I would want to do. 

If you have questions please ask, if you have ideas how to move forward, please tell :) All for now, I'll add comments to your comments and then finish for tonight.

Screenshots:
User List
![user_list](https://user-images.githubusercontent.com/729888/43359636-cf1b23d4-92a5-11e8-84fe-bb1db3724702.png)
Add Users
![add_user](https://user-images.githubusercontent.com/729888/43359679-a152471a-92a6-11e8-8145-8b88799b9fce.png)
Moderator Menu
![moderate_user](https://user-images.githubusercontent.com/729888/43359735-c25aff28-92a7-11e8-8d0b-8b506c9f5f37.png)
User Settings
![settings](https://user-images.githubusercontent.com/729888/43359759-87a18ee6-92a8-11e8-9326-146fd30090fa.png)

frequent commented:
@gnunicorn : AO-Änderung ~ ist das eine außerordentliche Änderung oder eine natürliche Weiterentwicklung von einer Initiative? Den Namen finde ich ein wenig unglücklich :)

GuidoDr commented:
I'm closing this old PR since I am doing some cleanup to get down to zreo PR.
